src/test/java/org/openrewrite/java/migrate/jakarta/DeprecatedCDIAPIsRemoved40Test.java

```diff
@@ -0,0 +1,106 @@
+            package sample.cdi;
+            
+            import jakarta.enterprise.event.Observes;
+            import jakarta.enterprise.inject.New;
+            import jakarta.enterprise.inject.spi.AnnotatedType;
+            import jakarta.enterprise.inject.spi.Bean;
+            import jakarta.enterprise.inject.spi.BeanManager;
+            import jakarta.enterprise.inject.spi.BeforeBeanDiscovery;
+            import jakarta.enterprise.inject.spi.Extension;
+            import java.util.Set;
+            
+            public class JakartaCdiMethods implements Extension {            
+            
+            	public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscovery, BeanManager beanManager) {
+            		System.out.println("SPI registered extention beforeBeanDiscovery has been fired");
+            		AnnotatedType<String> producerType = beanManager.createAnnotatedType(String.class);
+ 
+            		beforeBeanDiscovery.addAnnotatedType(producerType);									// Flag this one
+            		beforeBeanDiscovery.addAnnotatedType(producerType, "my unique id");				// Not this one
+            		beforeBeanDiscovery.addAnnotatedType(String.class, "my other unique id");	// Not this one
+            
+            		beanManager.createInjectionTarget(producerType);
+            
+            		beanManager.fireEvent(beforeBeanDiscovery);
+            
+            		Set<Bean<?>> myBeans = beanManager.getBeans("my precious beans");
+            		Bean myFavoriteBean = myBeans.stream().findFirst().get();
+            		myFavoriteBean.isNullable();
+            	}
+            }
+            """, """
+            package sample.cdi;
+            
+            import jakarta.enterprise.event.Observes;
+            import jakarta.enterprise.inject.New;
+            import jakarta.enterprise.inject.spi.AnnotatedType;
+            import jakarta.enterprise.inject.spi.Bean;
+            import jakarta.enterprise.inject.spi.BeanManager;
+            import jakarta.enterprise.inject.spi.BeforeBeanDiscovery;
+            import jakarta.enterprise.inject.spi.Extension;
+            import java.util.Set;
+            
+            public class JakartaCdiMethods implements Extension {
+            
+            	public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscovery, BeanManager beanManager) {
+            		System.out.println("SPI registered extention beforeBeanDiscovery has been fired");
+            		AnnotatedType<String> producerType = beanManager.createAnnotatedType(String.class);
+            
+            		beforeBeanDiscovery.addAnnotatedType(producerType, null);									// Flag this one
+            		beforeBeanDiscovery.addAnnotatedType(producerType, "my unique id");				// Not this one
+            		beforeBeanDiscovery.addAnnotatedType(String.class, "my other unique id");	// Not this one
+            
+            		beanManager.getInjectionTargetFactory().createInjectionTarget(producerType);
```

Comment:
I couldnt get this to work as well since if the arguments are right I was using the simpleName change and that worked..now we need to add different arguments to the methods as well. I updated the test

Reply:
Okay I'll look into both updates